default/python/AI/AIFleetMission.py

```diff
@@ -139,79 +135,53 @@ def _get_fleet_order_from_target(self, mission_type, target):
         for fid in other_fleets_here:
-            fleet_role = aistate.get_fleet_role(fid)
-            if fleet_role not in COMPATIBLE_ROLES_MAP[main_fleet_role]:
-                continue
-            fleet = universe.getFleet(fid)
-
-            if not fleet or fleet.systemID != system_id or len(fleet.shipIDs) == 0:
-                continue
-            if not (fleet.speed > 0 or main_fleet.speed == 0):  # TODO(Cjkjvfnby) Check this condition
-                continue
             fleet_mission = aistate.get_fleet_mission(fid)
-            do_merge = False
-            need_left = 0
-            if (main_fleet_role == MissionType.ORBITAL_DEFENSE) or (fleet_role == MissionType.ORBITAL_DEFENSE):
-                if main_fleet_role == fleet_role:
-                    do_merge = True
-            elif (main_fleet_role == MissionType.ORBITAL_INVASION) or (fleet_role == MissionType.ORBITAL_INVASION):
-                if main_fleet_role == fleet_role:
-                    do_merge = False  # TODO: could allow merger if both orb invaders and both same target
-            elif not fleet_mission and (main_fleet.speed > 0) and (fleet.speed > 0):
-                do_merge = True
-            else:
-                if not self.target and (main_fleet.speed > 0 or fleet.speed == 0):
-                    do_merge = True
-                else:
-                    target = fleet_mission.target.id if fleet_mission.target else None
-                    if target == target_id:
-                        info("Military fleet %d (%d ships) has same target as %s fleet %d (%d ships). Merging former "
-                             "into latter." % (fid, fleet.numShips, fleet_role, fleet_id, len(main_fleet.shipIDs)))
-                        # TODO: should probably ensure that fleetA has aggression on now
-                        do_merge = float(min(main_fleet.speed, fleet.speed))/max(main_fleet.speed, fleet.speed) >= 0.6
-                    elif main_fleet.speed > 0:
-                        neighbors = aistate.systemStatus.get(system_id, {}).get('neighbors', [])
-                        if target == system_id and target_id in neighbors and self.type == MissionType.SECURE:
-                            # consider 'borrowing' for work in neighbor system  # TODO check condition
-                            need_left = 1.5 * sum(aistate.systemStatus.get(nid, {}).get('fleetThreat', 0)
-                                                  for nid in neighbors if nid != target_id)
-                            fleet_rating = CombatRatingsAI.get_fleet_rating(fid)
-                            if need_left < fleet_rating:
-                                do_merge = True
-            if do_merge:
-                FleetUtilsAI.merge_fleet_a_into_b(fid, fleet_id, need_left,
-                                                  context="Order %s of mission %s" % (context, self))
-        return
+            if fleet_mission.type != self.type or fleet_mission.target != self.target:
```

Comment:
FYI: you can replace

```python
fleet_mission.type != self.type or fleet_mission.target != self.target
``` 
with
```python
(fleet_mission.type, fleet_mission.target ) != (self.type, self.target)
```

Sometimes this approach can improve readability.